grpc-server/template_test.go

```diff
@@ -50,7 +67,26 @@ func TestCreateTemplate(t *testing.T) {
 	}{
 		"SuccessfullTemplateCreation": {
 			args: args{
-				db:        mock.DB{},
+				db: mock.DB{
```

Comment:
Why the switch from `mockDB{}` to overriding `CreateTemplateFunc`? From what I can tell the expected behavior appears to be the same.

Reply:
No, Not exactly, the definition of `CreateTemplateFunc` will depend on the test case I would like to test.

Reply:
Done